spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java

```diff
@@ -64,6 +65,18 @@ public void verifyShouldUseProxyTarget() throws Exception {
 		verify(this.dateService, times(1)).getDate(matchesAnyBoolean());
 	}
 
+	@Test
+	public void verifyShouldUseProxyTargetWithMethodArguments() throws Exception {
+		Long d1 = this.dateService.getRandomDate(1234L);
+		Thread.sleep(200);
+		Long d2 = this.dateService.getRandomDate(1234L);
+		assertThat(d1).isEqualTo(d2);
+		verify(this.dateService, times(1)).getDate(false);
+		verify(this.dateService, times(1)).getDate(matchesFalse());
+		verify(this.dateService, times(1)).getDate(matchesAnyBoolean());
```

Comment:
These three verifications are verifying the wrong method. They should be checking that `getRandomDate(long)` has only been called once:

```java
verify(this.dateService, times(1)).getRandomDate(1234L);
```

With this change in place, the test passes.

Reply:
Interestingly enough this failed at calling the actual method (so I didn't even get to the assertions). Will fix the assertions and see what happens.